### ISSUE TITLE
Make the Kubernetes Settings page more consistent when any setting changes

### ISSUE THREAD

**Issue by ericpromislow** [rancher-sandbox/rancher-desktop#1643]
### Problem Description

Currently changes to some of the settings are deferred: a blue box appears at the bottom specifying that the cluster must be reset for the change to take effect. Others, namely the container-runtime and enable-k8s checkboxes, trigger a yes/no dialog box, and if the user chooses 'yes' the cluster is immediately reset.

The `enable-traefik` lives in its own world, involving neither the blue-box nor an immediate change.

All the changes should go through the delayed blue-box. 

For one thing, it's currently impossible to go in one step from (k8s-on, moby) to (k8s-off, containerd).

### Proposed Solution

All desired changes in the K8s Settings page are reflected in the blue box at the bottom of the screen, and don't take effect until the user either presses the "Reset Kubernetes" button or shuts the app down and restarts it.
### Preflight Checklist

- [X] I have searched the [issue tracker](https://github.com/rancher-sandbox/rancher-desktop/issues) for a feature request that matches the one I want to file, without success.

**Comment by jandubois** [comment]
> All desired changes in the K8s Settings page are reflected in the blue box at the bottom of the screen, and don't take effect until the user either presses the "Reset Kubernetes" button or shuts the app down and restarts it.

The problem though is that this is a settings **page**, so you can navigate away from it without either applying or canceling your changes. This should be a modal dialog instead.

**Comment by ericpromislow** [comment]
Not a big problem, because the preferences config always contains the current desired values. The blue box reflects differences between the desired and current values, even if you navigate away and later return.

The most annoying thing I found is when I leave the page and return, the `enable traefik` button blinks from on to off when the pref value is off.

**Comment by jandubois** [comment]
> Not a big problem, because the preferences config always contains the current desired values. The blue box reflects differences between the desired and current values, even if you navigate away and later return.

Yes, but it is a kludge. It is almost always a mistake when the user moves away from the setting change page without either resetting or applying the changes. The internal app state may be consistent with the UI, but the state in the users mind may no longer match the state of the app.

This is of course worse because the blue box is by default not visible because the settings page has a scrollbar and the blue bar is outside the visible area when the window has the default size.

Finally, there are bugs in the current implementation: the blue box has an `x` button to dismiss it, but the settings on the page do not revert to their original setting. Further changes to the setting will not bring back the blue bar, so you can no longer tell if the slider represents a desired or actual state.

In summary, I'm of the firm opinion that a settings dialog for a real app should be modal and have no scrollbar.

**Comment by rak-phillip** [comment]
I never liked the solution of using alerts at the bottom of the page or the pattern of Reseting Kubernetes to apply changes. I feel like this page has also grown to encompass more than just Kubernetes Settings.. Perhaps, we can be more intentioned with our change and reintroduce our discussion of redesigning this section?

**Comment by ericpromislow** [comment]
+1 on the notification being out of the way and an unnoticeable color
+1 on that this page shouldn't have scroll bars

I'm not sure if we need a modal dialog to change these though. But changes should be additive (like switching engines and kubernetes on/off at the same time), and the UI should be consistent.

**Comment by rak-phillip** [comment]
I've started a discussion with a proposal and some other points to consider

#1654

**Comment by rak-phillip** [comment]
@jandubois I'm closing this issue because I feel it was addressed in #2373
